lib/components/form/batch-settings.tsx

```diff
@@ -216,6 +225,7 @@ class BatchSettings extends Component<{
 const mapStateToProps = (state: any) => ({
   config: state.otp.config,
   currentQuery: state.otp.currentQuery,
+  modeOptions: state.otp.config.modes.modeOptions,
```

Comment:
I had a total crash when I tried to load a config without the new options. I know we need the `requiredModes` for the new stuff to work, but what if we set a default here and then extracted the requiredModes from the string if they weren't explicitly specified? I know this would probably result in weird behavior but I think the backwards compatibility is valuable, and it gives a nice "it just works" feeling

Reply:
Okay, some big changes to add in defaults similar to old behavior. (which already was not great and a bit glitchy, mind you)